### PR TITLE
Extend CodeQL scan with security-and-quality queries 

### LINKED ISSUE #648: Extend the scope of CodeQL scan
The `codeql-analysis.yml` workflow runs a CodeQL CI workflow executing a set of security queries (~44) to prevent security vulnerabilities. This workflow blocks PRs where vulnerabilities are detected.

CodeQL provides [extended query suites](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs) that provide additional coverage for security and code quality checks. The goal of this issue is to introduce the `security-and-quality` CodeQL suite in the CI workflow to detect bugs and code smells and thus raise code quality level in EDC. Security-and-quality query suite is built into CodeQL and available for use.

Introducing this new set of queries leads to ~ 200 alerts in the current EDC codebase. Alerts with the top 2 severities (Error/Critical, Warning/High) should be analysed (~ 20 alerts). Low-hanging fruits should be fixed right away, while expensive alerts must be suppressed with a corresponding issue for solving them separately. Alerts of lower severities can stay suppressed.

Tasks
- [ ] Use the `security-and-quality` CodeQL suite in the CodeQL workflow
- [ ] Agree on minimal alert severity to block PRs
- [ ] Fix or suppress existing alerts reported by CodeQL (create separate items to fix these later if needed)

### PR DESCRIPTION
**Pull request by Izzzu**
This PR extends the CodeQL scan to perform additional quality checks on the code.

- Change in the codeql-analysis.yml file to run extended query suite of CodeQL scan (sequirity-and-quality). It runs the scan only on the production code.
- Documentation file for CodeQL.
- Solving some alerts found by CodeQL with severity: Error/Critical, Medium,High/Warning. Alerts with severity Low/Note were suppressed. 
- Alerts that are not easily fixable can be suppressed for now and solved in separate issues.

Note: 
- not relevant alerts can be suppressed only by users with Write permissions 
- alerts with Low/Note severity do not break the build
- after merging this PR CodeQL scan will take less time: 8-12 min instead of up to 22 min currently

closes #648

### CODE REVIEW

File: extensions/ion/ion-core/src/main/java/org/eclipse/dataspaceconnector/ion/spi/request/InputValidator.java [hunk 1]
```diff
@@ -9,10 +9,6 @@
 
 public class InputValidator {
     public static void validateEs256kOperationKey(JWK key, String visibility) {
-        var allowedProperties = new HashSet<>(Set.of("kty", "crv", "x", "y"));
```
Comment (paullatzelsperger) [inline extensions/ion/ion-core/src/main/java/org/eclipse/dataspaceconnector/ion/spi/request/InputValidator.java:12]: Just out of curiosity: why did CodeQL complain here?

File: extensions/jdk-logger-monitor/src/main/java/org/eclipse/dataspaceconnector/logger/LoggerMonitor.java [hunk 2]
```diff
@@ -60,8 +61,10 @@ private void log(final Supplier<String> supplier, final Level level, final Throw
         if (errors == null || errors.length == 0) {
             LOGGER.log(level, supplier);
         } else {
-            Arrays.stream(errors).forEach(error -> LOGGER.log(level, supplier.get(), error));
+            String logMessage = Optional.ofNullable(supplier.get())
```
Comment (paullatzelsperger) [inline extensions/jdk-logger-monitor/src/main/java/org/eclipse/dataspaceconnector/logger/LoggerMonitor.java:64]: Just out of curiosity: why did CodeQL complain here?

**Review by paullatzelsperger — APPROVED** [review]

File: extensions/jdk-logger-monitor/src/main/java/org/eclipse/dataspaceconnector/logger/LoggerMonitor.java (hunk 2 above)
Comment (Izzzu) [reply extensions/jdk-logger-monitor/src/main/java/org/eclipse/dataspaceconnector/logger/LoggerMonitor.java:64]: It complained about Log injection
https://owasp.org/www-community/attacks/Log_Injection

File: extensions/ion/ion-core/src/main/java/org/eclipse/dataspaceconnector/ion/spi/request/InputValidator.java (hunk 1 above)
Comment (Izzzu) [reply extensions/ion/ion-core/src/main/java/org/eclipse/dataspaceconnector/ion/spi/request/InputValidator.java:12]: Here the alert was: [**Container contents are never accessed**](https://codeql.github.com/codeql-query-help/java/java-unused-container/)

**Comment by Izzzu** [comment]
> could you have the LRUCache implement Cloneable and throw UnsupportedOperationException and we can merge this?

@jimmarino Thanks for pointing it out, I missed this alert as it came as new. I applied your suggestion

**Pull request merged**